### ISSUE TITLE
Amazon S3 REST services not working.

### ISSUE THREAD

**Issue by fusionstrings** [strongloop/loopback-component-storage#18]
Hi,
This issue is related to https://github.com/strongloop/loopback-storage-service/issues/16.
I am trying to upload to amazon S3 using REST API. My keys have full admin access.
I am able to get files but any kind of post is not working.

I tried running associated example. The `GET` method is working but any other operation is not working.

upload response is returning 200 but nothing is uploaded. 
![image](https://cloud.githubusercontent.com/assets/450624/3426175/ccb7f674-001b-11e4-9eb5-7dd5eda6cdf9.png)
Not just this `POST' response is returning uploaded file but subsequent get call returns actual list of files which is sans earlier uploaded files.

![image](https://cloud.githubusercontent.com/assets/450624/3426164/904ffed4-001b-11e4-9dfb-a6c14c0b16b9.png)

![image](https://cloud.githubusercontent.com/assets/450624/3426169/acea6f52-001b-11e4-8f96-86fdc9ca0751.png)

Kindly do the needful.

** Update **
I did some research and found out that if files don't have space in their name are in fact getting uploaded but in example provided `GET` request immediately followed by `POST` doesn't return them (as well) in response. If page is refreshed then it returns correct response.

So basically it's here we are stuck apart from dynamic container creation.
IMHO loopback-storage REST services are not behaving as documented. I should have been able to upload the files using postman but I am not. So may be either I am missing something or the documentation.

Kindly guide me accordingly. Thanks.

**Comment by richardpringle** [comment]
@fusionstrings, did you ever get this working? I'm trying to close old issues if possible.

**Comment by fusionstrings** [comment]
Sorry @richardpringle , I indeed was able to solve it. I apologies for not closing it.

**Comment by richardpringle** [comment]
No problem @fusionstrings, glad everything worked out!
